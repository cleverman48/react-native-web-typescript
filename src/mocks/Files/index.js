const path = require('path');

module.exports = {
  process: (_, filename) =>
    `module.exports = '${JSON.stringify(path.basename(filename))}';`,
};
