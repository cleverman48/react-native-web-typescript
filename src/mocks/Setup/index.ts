import 'react-native-gesture-handler/jestSetup';

jest.useFakeTimers();

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => undefined;
  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

export const mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  push: jest.fn(),
  popToTop: jest.fn(),
};
jest.mock('@react-navigation/native', () => {
  return {
    useRoute: () => jest.fn(),
    useNavigation: () => ({
      goBack: mockNavigation.goBack,
      navigate: mockNavigation.navigate,
      push: mockNavigation.push,
      popToTop: mockNavigation.popToTop,
    }),
  };
});
