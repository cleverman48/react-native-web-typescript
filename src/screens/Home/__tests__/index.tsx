import React from 'react';
import {Home} from '..';
import renderer from 'react-test-renderer';
import {mockNavigation} from '../../../mocks/Setup';

it('should render correctly', () => {
  const component = renderer.create(<Home />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('navigates to details', () => {
  const component = renderer.create(<Home />);
  const button = component.root.findByProps({testID: 'details'});
  button.instance.props.onPress();
  expect(mockNavigation.navigate).toBeCalledWith('Details', {data: 'ðŸ¤ª'});
});
