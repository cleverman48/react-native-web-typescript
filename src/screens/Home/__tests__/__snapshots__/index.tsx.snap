// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home Screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="details"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Details
      </Text>
    </View>
  </View>
</View>
`;
